Source/SwiftLintBuiltInRules/Rules/Idiomatic/OneDelarationPerFileRule.swift

```diff
@@ -0,0 +1,118 @@
+import Foundation
+import SwiftSyntax
+
+@SwiftSyntaxRule
+struct OneDelarationPerFileRule: Rule {
+    var configuration = SeverityConfiguration<Self>(.error)
+
+    static let description = RuleDescription(
+        identifier: "one_declaration_per_file",
+        name: "One Declaration Per File",
+        description: "One declaration per file is allowed, extensions are an exception",
```

Comment:
If no specific reason is given in `violations.append(...)` this description will be the violation message. I'd pick something shorter like "Only a single declaration is allowed in a file".